provider/go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/pulumi/pulumi-auth0/provider/v2
 go 1.21
 
 replace (
+	github.com/auth0/terraform-provider-auth0 => ../upstream
 	github.com/auth0/terraform-provider-auth0/shim => ./shim
```

Comment:
We don't want to mix a shim with an upstream, since it adds upgrade complexity. Can we move the shim into `upstream/shim` via another patch?

Reply:
@iwahbe could you explain a bit more, I'd like to better understand this principle.  thx

Reply:
@EronWright we use a `provider/shim/` module to get access to functions behind `internal` in the upstream provider. This means that we have an additional `go.mod` that we need to update and keep in sync. We reduce the number of go modules by moving the shim into the `upstream` with a patch.

This reduces the number of places where we need to update a `replace` for testing and building the provider.

Reply:
It also has the nice benefit of ensuring that we are compiling against the patched upstream, since compiling against the raw provider will error: the raw provider doesn't expose a `shim` package.